File: Golang/RotateList/RotateList_test.go
package RotateList

import (
	"testing"
	"reflect"
)

func TestRotateRight(t *testing.T) {

	if rotateRight(nil, 1) == nil {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	// Expected: 1   -->   1
	if reflect.DeepEqual(rotateRight(&ListNode{
		Val: 1,
	}, 1), &ListNode{
		Val: 1,
	}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	// Expected: 1 -> 2   -->   1 -> 2
	if reflect.DeepEqual(rotateRight(&ListNode{
		Val: 1,
		Next: &ListNode{
			Val: 2,
		},
	}, 0), &ListNode{
		Val: 1,
		Next: &ListNode{
			Val: 2,
		},
	}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	// Expected: 1 -> 2 -> 3   --->   3 -> 1 -> 2
	if reflect.DeepEqual(rotateRight(&ListNode{
		Val: 1,
		Next: &ListNode{
			Val: 2,
			Next: &ListNode{
				Val: 3,
			},
		},
	}, 1), &ListNode{
		Val: 3,
		Next: &ListNode{
			Val: 1,
			Next: &ListNode{
				Val: 2,
			},
		},
	}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

	// Expected: 1 -> 2 -> 3 -> 4 -> 5   -->   4 -> 5 -> 1 -> 2 -> 3
	if reflect.DeepEqual(rotateRight(&ListNode{
		Val: 1,
		Next: &ListNode{
			Val: 2,
			Next: &ListNode{
				Val: 3,
				Next: &ListNode{
					Val: 4,
					Next: &ListNode{
						Val: 5,
					},
				},
			},
		},
	}, 2), &ListNode{
		Val: 4,
		Next: &ListNode{
			Val: 5,
			Next: &ListNode{
				Val: 1,
				Next: &ListNode{
					Val: 2,
					Next: &ListNode{
						Val: 3,
					},
				},
			},
		},
	}) {
		t.Log("pass")
	} else {
		t.Error("failed")
	}

}

File: Python/RotateListTest.py
from unittest import TestCase

from RotateList import RotateList, ListNode


class TestRotateList(TestCase):
    def test_rotateRight(self):
        rl = RotateList()

        self.assertIsNone(rl.rotateRight(None, 1))

        # Expected: 1
        tmp = ListNode(1)
        self.assertEqual(rl.rotateRight(ListNode(1), 1), tmp)

        # Expected: 1 -> 2
        node12 = ListNode(1)
        node12.next = ListNode(2)
        tmp.next = ListNode(2)
        self.assertEqual(rl.rotateRight(node12, 0), tmp)

        # Expected: 3 -> 1 -> 2
        node123 = ListNode(1)
        node123.next = ListNode(2)
        node123.next.next = ListNode(3)
        tmp = ListNode(3)
        tmp.next = ListNode(1)
        tmp.next.next = ListNode(2)
        self.assertEqual(rl.rotateRight(node123, 1), tmp)

        node12345 = ListNode(1)
        node12345.next = ListNode(2)
        node12345.next.next = ListNode(3)
        node12345.next.next.next = ListNode(4)
        node12345.next.next.next.next = ListNode(5)
        tmp = ListNode(4)
        tmp.next = ListNode(5)
        tmp.next.next = ListNode(1)
        tmp.next.next.next = ListNode(2)
        tmp.next.next.next.next = ListNode(3)
        self.assertEqual(rl.rotateRight(node12345, 2), tmp)
